File: LEVEL2/더 맵게.java
import java.io.*;
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> pq = new PriorityQueue();
        
        for(int i : scoville) {
            pq.offer(i);
        }
        
        while(pq.peek() <= K) {
            if(pq.size() == 1) {
                return -1;
            }
            
            int x = pq.poll();
            int y = pq.poll();
            int res = x + (y * 2);
            
            pq.offer(res);
            answer++;
        }
        
        return answer;
    }
}

File: LEVEL2/더 맵게.cpp
#include<queue>
#include<vector>
#include<algorithm>
using namespace std;

int solution(vector<int> scoville, int K) {
    int answer = 0;
    priority_queue<int, vector<int>, greater<int>> pq;

    for(int i : scoville) {
        pq.push(i);
    }
    
    while(pq.top() <= K) {
        if(pq.size() == 1) {
            return -1;
        }
        
        int x = pq.top();
        pq.pop();
        int y = pq.top();
        pq.pop();
        x += (y * 2);
        
        pq.push(x);
        answer++;
    }
    
    return answer;
}